src/main/kotlin/de/triplet/gradle/play/BootstrapTask.kt

```diff
@@ -1,63 +1,91 @@
 package de.triplet.gradle.play
 
+import com.google.api.services.androidpublisher.AndroidPublisher
+import de.triplet.gradle.play.internal.ImageType
+import de.triplet.gradle.play.internal.LISTING_PATH
+import de.triplet.gradle.play.internal.ListingDetail
+import de.triplet.gradle.play.internal.PlayPublishTaskBase
 import org.gradle.api.tasks.TaskAction
 import java.io.File
+import java.net.URL
 
-open class BootstrapTask : PlayPublishTask() {
+open class BootstrapTask : PlayPublishTaskBase() {
     lateinit var outputFolder: File
 
     @TaskAction
-    fun bootstrap() {
-        publish()
-
-        bootstrapListing()
-        bootstrapWhatsNew()
-        bootstrapAppDetails()
+    fun bootstrap() = read { editId ->
+        bootstrapListing(editId)
+        bootstrapWhatsNew(editId)
+        bootstrapAppDetails(editId)
     }
 
-    private fun bootstrapListing() {
-        val listings = edits.listings()
+    private fun AndroidPublisher.Edits.bootstrapListing(editId: String) {
+        val listings = listings()
                 .list(variant.applicationId, editId)
                 .execute()
                 .listings ?: return
 
-        listings.forEach { listing ->
-            val listingDir = outputFolder.validSubFolder(listing.language, LISTING_PATH) ?: return@forEach
+        for (listing in listings) {
+            val rootDir = File(outputFolder, "${listing.language}/$LISTING_PATH")
+
+            fun downloadMetadata() {
+                fun String.write(detail: ListingDetail) = write(rootDir, detail)
+
+                listing.fullDescription.write(ListingDetail.FULL_DESCRIPTION)
+                listing.shortDescription.write(ListingDetail.SHORT_DESCRIPTION)
+                listing.title.write(ListingDetail.TITLE)
+                listing.video.write(ListingDetail.VIDEO)
+            }
+
+            fun downloadImages() {
+                for (type in ImageType.values()) {
+                    val images = images()
+                            .list(variant.applicationId, editId, listing.language, type.fileName)
+                            .execute()
+                            .images ?: continue
+                    val imageDir = File(rootDir, type.fileName)
 
-            ImageTypes.values().forEach { imageType ->
-                val images = edits.images()
-                        .list(variant.applicationId, editId, listing.language, imageType.fileName)
-                        .execute()
-                        .images
-                imageType.saveImages(listingDir, images)
+                    for (image in images) {
```

Comment:
The for loop is actually shorter than the `forEach` one. Also, a for loop is clearer IMO and just because something exists doesn't mean you have to use it. Dunno, what'd you think?